network-config-analyzer/NetworkConfigQuery.py

```diff
@@ -614,12 +615,174 @@ def exec(self):
 
         return QueryAnswer(True, f'{self.name1} and {self.name2} are semantically equivalent.')
 
+class SemanticDiffQuery(TwoNetworkConfigsQuery):
+    """
+    Produces a report of newly allowed and newly denied connections
+    """
+    class SingleDiff:
```

Comment:
Better make this class a dataclass (like `PolicyConnections` in `NetworkPolicy.py`)

Reply:
good point